include/deal.II/lac/scalapack.h

```diff
@@ -189,6 +189,163 @@ class ScaLAPACKMatrix : protected TransposeTable<NumberType>
 
+  /**
+   * Transposing assignment: <i>A = B<sup>T</sup></i>
+   *
+   * The matrices <tt>A</tt> and <tt>B</tt> must have the same process grid
+   *
+   * Following alignment conditions have to be fulfilled: MB_A = NB_B and NB_A = MB_B
+   */
+  void copy_transposed(const ScaLAPACKMatrix<NumberType> &B);
+
+  /**
+   * Matrix-addition:
+   *
+   * <i>A = a A + b op(B)</i>
+   *
+   * if(transpose_B)  <i>op(B) = B<sup>T</sup></i>
+   *
+   * else  <i>op(B) = B</i>
+   *
+   * The matrices <tt>A</tt> and <tt>B</tt> must have the same process grid
+   *
+   * Following alignment conditions have to be fulfilled:
+   *
+   * | transpose_B |          Block Sizes         |
+   * | :---------: | :--------------------------: |
+   * |   false     | MB_A = MB_B <br> NB_A = NB_B |
+   * |   true      | MB_A = NB_B <br> NB_A = MB_B |
+   */
+  void add(const ScaLAPACKMatrix<NumberType> &B,
+           const NumberType a=0.,
+           const NumberType b=1.,
+           const bool transpose_B=false);
+
+  /**
+   * Matrix-addition:
+   *
+   * <i>A += b B</i>
+   *
+   * The matrices <tt>A</tt> and <tt>B</tt> must have the same process grid
+   *
+   * Following alignment conditions have to be fulfilled: MB_A = MB_B and NB_A = NB_B
+   */
+  void add(const NumberType b,
+           const ScaLAPACKMatrix<NumberType> &B);
+
+  /**
+   * Matrix-addition:
+   *
+   * <i>A += b B<sup>T</sup></i>
+   *
+   * The matrices <tt>A</tt> and <tt>B</tt> must have the same process grid
+   *
+   * Following alignment conditions have to be fulfilled: MB_A = NB_B and NB_A = MB_B
+   */
+  void Tadd(const NumberType b,
+            const ScaLAPACKMatrix<NumberType> &B);
+
+  /**
+   * Matrix-matrix-multiplication:
+   *
+   * The operations based on the input parameters and the alignment conditions are summarized in the following table:
+   *
+   * | transpose_A | transpose_B |                   Block Sizes                 |                     Operation                    |
+   * | :---------: | :---------: | :-------------------------------------------: | :----------------------------------------------: |
+   * | false       |   false     | MB_A = MB_C <br> NB_A = MB_B <br> NB_B = NB_C |             <i>C = a A * B + b C</i>             |
+   * | false       |   true      | MB_A = MB_C <br> NB_A = NB_B <br> MB_B = NB_C |      <i>C = a A * B<sup>T</sup> + b C</i>        |
+   * | true        |   false     | MB_A = MB_B <br> NB_A = MB_C <br> NB_B = NB_C |        <i>C = a A<sup>T</sup> * B + b C</i>      |
+   * | true        |   true      | MB_A = NB_B <br> NB_A = MB_C <br> MB_B = NB_C | <i>C = a A<sup>T</sup> * B<sup>T</sup> + b C</i> |
+   *
+   * It is assumed that <tt>A</tt> and <tt>B</tt> have compatible sizes and that
+   * <tt>C</tt> already has the right size.
+   *
+   * The matrices <tt>A</tt>, <tt>B</tt> and <tt>C</tt> must have the same process grid.
+   */
+  void mult(ScaLAPACKMatrix<NumberType> &C,
```

Comment:
@dealii/dealii this is kind of `smult` (scale matrix and add multiplication). For consistency with vectors, i would probably move two numbers in front i.e. `(const NumberType s, const NumberType a, const Matrix....)`.
But then it would be inconsistent with `FullMatrix`

```
void mmult (FullMatrix< number2 > &C, const FullMatrix< number2 > &B, const bool adding=false) const;
```

hm...

Reply:
We used to have similar functions in some of the vector classes (e.g [here](https://www.dealii.org/8.5.0/doxygen/deal.II/classTrilinosWrappers_1_1VectorBase.html#ab0d14dd0a857a51f18562e84a989710c)), so I would say that the factors and matrices should be ordered according to the operation along with the input matrices, even if that means that we lose the default values. Does a default value of 0 for `b` make sense anyway? This then just becomes an assignment operation.